en/debug/plotting_realtime_uorb_data.md

```diff
@@ -0,0 +1,118 @@
+# Plotting uORB Topic Data in Real Time using PlotJuggler
+
+This topic shows how you can graph the "live" values of [uORB topics](../msg_docs/index.md) (in real time) using [PlotJuggler](../log/flight_log_analysis.md#plotjuggler) and the _uXRCE-DDS Agent_.
+
+This technique uses PX4 [uXRCE-DDS](../middleware/uxrce_dds.md) middleware to export uORB topics as ROS2 topics, which can then be read and plotted by PlotJuggler as they change (PlotJuggler cannot directly read uORB topics, but the values of the corresponding ROS 2 topics are the same).
+
+The video below demonstrates this for a simulated vehicle — the approach works equally well on real hardware.
+
+<video src="../../assets/debug/realtime_debugging/realtime_debugging.mp4" width="720" controls></video>
+
+## Prerequisites
+
+Follow the [ROS 2 Installation & Setup](../ros2/user_guide.md#installation-setup) instructions in the _ROS2 user guide_ to install:
+
+- ROS 2
+- [Micro XRCE-DDS Agent](../ros2/user_guide.md#setup-micro-xrce-dds-agent-client)
+- [PX4/px4_msgs](https://github.com/PX4/px4_msgs): PX4/ROS2 shared message definitions.
+- PX4 source code and build the simulator.
+
+  ::: tip
+  If you're using real hardware instead of the simulator, you will only need PX4 source code if you need to change the set of topics that are published to ROS 2 (only a subset of uORB topics are published by default).
+  :::
+
+You will also need to install:
+
+- [PlotJuggler for ROS2](https://github.com/facontidavide/PlotJuggler)
+
+  ::: tip
+  Use the Debian packages (the snap files are not supported).
+  :::
+
+## Usage
+
+First we need to build a ROS 2 workspace that includes the `px4_msgs` that correspond to the PX4 build to be monitored, and then launch PlotJuggler from within that workspace.
+This allows ROS 2 and PlotJuggler to interpret the messages.
+If you're using unmodified PX4, the definitions from [PX4/px4_msgs](https://github.com/PX4/px4_msgs) can be used.
+
+::: info
+This is the same process as covered in [Build ROS 2 Workspace](../ros2/user_guide.md#build-ros-2-workspace) in _ROS 2 Installation & Setup_.
+:::
+
+Assuming your ROS 2 workspace is `~/ros2_ws/`, fetch and build the `px4_msgs` package in a terminal as shown:
+
+```sh
+cd ~/ros2_ws/src/
+git clone https://github.com/PX4/px4_msgs.git
+cd ..
+colcon build
+source install/setup.bash
+```
+
+Then run PlotJuggler by entering the following commands in a terminal:
+
+```sh
+ros2 run plotjuggler plotjuggler
+```
+
+To start sending ROS 2 topics from PX4 the uXRCE-DDS **client** has to be running on PX4, and the `MicroXRCEAgent` has to be running on the same computer as PlotJuggler.
+If you're using the Simulator then the client is started automatically.
+Assuming you are using the [Gazebo](../sim_gazebo_gz/index.md) simulator, you might start it in another terminal like this:
+
+```sh
+make px4_sitl gz_x500
+```
+
```

Comment:
I added a simulator example here. Never sure how much you should say, but if we want this to be useful for "dummies" then it doesn't hurt to spell out at least one example that should work out of the box by following instructions end to end.

Reply:
good idea!